packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -151,25 +154,45 @@ async function main() {
       return;
     }
 
-    if (hardhatArguments.config === undefined && !isCwdInsideProject()) {
+    // ATTENTION! DEPRECATED CODE!
+    // The command `npx hardhat`, when used to create a new Hardhat project, will be removed with Hardhat V3.
+    // It will become `npx hardhat init`.
+    // The code marked with the tag #INIT-DEP can be deleted after HarhatV3 is out.
+
+    // Create a new Hardhat project
+    if (parsedTaskName === "init") {
+      return await createNewProject();
+    }
+    // #INIT-DEP - START OF DEPRECATED CODE
+    else {
       if (
-        process.stdout.isTTY === true ||
-        process.env.HARDHAT_CREATE_JAVASCRIPT_PROJECT_WITH_DEFAULTS !==
-          undefined ||
-        process.env.HARDHAT_CREATE_TYPESCRIPT_PROJECT_WITH_DEFAULTS !==
-          undefined
+        parsedTaskName === undefined &&
+        hardhatArguments.config === undefined &&
+        !isCwdInsideProject()
       ) {
-        await createProject();
+        await createNewProject();
+
+        // Warning for Hardhat V3 deprecation
+        console.warn(
+          chalk.yellow.bold("\n\nDEPRECATION WARNING\n\n"),
+          chalk.yellow(
+            `Initializing a project with ${chalk.white.italic(
+              "npx hardhat"
+            )} is deprecated and will be removed in the future.\n`
+          ),
+          chalk.yellow(
+            `Please use ${chalk.white.italic("npx hardhat init")} instead.\n\n`
+          )
+        );
+
         return;
       }
+    }
+    // #INIT-DEP - END OF DEPRECATED CODE
 
-      // Many terminal emulators in windows fail to run the createProject()
-      // workflow, and don't present themselves as TTYs. If we are in this
-      // situation we throw a special error instructing the user to use WSL or
-      // powershell to initialize the project.
-      if (process.platform === "win32") {
-        throw new HardhatError(ERRORS.GENERAL.NOT_INSIDE_PROJECT_ON_WINDOWS);
-      }
+    // Tasks are only allowed inside a Hardhat project (except the init task)
+    if (!isCwdInsideProject()) {
```

Comment:
I think we should do this here:

```suggestion
    if (hardhatArguments.config === undefined && !isCwdInsideProject()) {
```

Otherwise you can't do something like:

1. Install Hardhat (but don't create a project)
2. Run `npx hardhat node --config /path/to/some/config/in/another/directory`

This works in the current version of Hardhat, but it doesn't in this PR. It wouldn't surprise me if people are relying on this behavior for some use cases (like starting a node anywhere without having to cd into a project).

Reply:
done